netbox/dcim/forms/model_forms.py

```diff
@@ -291,7 +291,11 @@ class DeviceTypeForm(NetBoxModelForm):
     default_platform = DynamicModelChoiceField(
         label=_('Default platform'),
         queryset=Platform.objects.all(),
-        required=False
+        required=False,
+        selector=True,
+        query_params={
+            'available_for_manufacturer': '$manufacturer',
+        }
```

Comment:
We can achieve this without introducing a dedicate filter.
```suggestion
        query_params={
            'manufacturer_id': ['$manufacturer', 'null'],
        }
```